packages/core/src/pages/s.tsx

```diff
@@ -73,15 +80,46 @@ function Page({ page: { sections, settings }, globalSections }: Props) {
   const title = 'Search Results'
   const { description, titleTemplate } = storeConfig.seo
 
+  const { page, sort, term, selectedFacets } = searchParams
+  const itemsPerPage = settings?.productGallery?.itemsPerPage ?? ITEMS_PER_PAGE
+
+  const { data: pageProductGalleryData } = useProductGalleryQuery({
+    term,
+    sort,
+    selectedFacets,
+    itemsPerPage,
+  })
+
+  const { productsPerPage } = usePageProductsQuery({
+    page,
+    term,
+    sort,
+    selectedFacets,
+    itemsPerPage,
+  })
+
   if (!searchParams) {
     return null
   }
 
+  const server = {
+    title,
+    searchTerm: searchParams.term ?? undefined,
+  } as SearchPageContextType
+
+  const context = {
+    data: {
+      ...server,
+      ...pageProductGalleryData,
+      productsPerPage,
```

Comment:
Why no deepmerge here?

Reply:
server data in search page is fixed (`title` and `searchTerm`), since `pageProductGalleryData` brings different values I think doesn't worth to deepmerge here, what do you think?

Reply:
We may change that in the future, but I think it's safe for now.